[Dist CI] Move rest of distributed tests to their own CI job

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64413
* __->__ #64253

Follow up to D30496178 to move the rest of distributed tests to their own jobs for Linux GHA.

Differential Revision: [D30662999](https://our.internmc.facebook.com/intern/diff/D30662999/)

rohan-varma committed:
[Dist CI] Move rest of distributed tests to their own CI job

Follow up to D30496178 to move the rest of distributed tests to their own jobs for Linux GHA.

Differential Revision: [D30662999](https://our.internmc.facebook.com/intern/diff/D30662999/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Dist CI] Move rest of distributed tests to their own CI job"


Follow up to D30496178 to move the rest of distributed tests to their own jobs for Linux GHA.

Differential Revision: [D30662999](https://our.internmc.facebook.com/intern/diff/D30662999/)

[ghstack-poisoned]

walterddr reviewed (approved):
looks good. i think there might be a much easier way to define this once #64246 is merged.

rohan-varma committed:
Update on "[Dist CI] Move rest of distributed tests to their own CI job"


Follow up to D30496178 to move the rest of distributed tests to their own jobs for Linux GHA.

Differential Revision: [D30662999](https://our.internmc.facebook.com/intern/diff/D30662999/)

[ghstack-poisoned]

rohan-varma commented:
We can see the appropriate tests running in the CI jobs such as:
https://github.com/pytorch/pytorch/pull/64253/checks?check_run_id=3489530778
https://github.com/pytorch/pytorch/pull/64253/checks?check_run_id=3489506459
https://github.com/pytorch/pytorch/pull/64253/checks?check_run_id=3489506459
